src/main/java/com/conveyal/r5/analyst/TravelTimeSurfaceComputer.java

```diff
@@ -0,0 +1,179 @@
+
+            int[] travelTimesToTargets = linkedDestinations.eval(sr::getTravelTimeToVertex, offstreetTravelSpeedMillimetersPerSecond).travelTimes;
+            for (int target = 0; target < travelTimesToTargets.length; target++) {
+                int x = target % request.width;
+                int y = target / request.width;
+
+                final int travelTimeMinutes =
+                        travelTimesToTargets[target] == RaptorWorker.UNREACHED ? RaptorWorker.UNREACHED : travelTimesToTargets[target] / 60;
+                // the frontend expects percentiles of travel time. There is no variation in nontransit travel time so
+                // just replicate the same number repeatedly. This could be improved, but at least it will compress well.
+                // int divide (floor) used below as well. TODO is this wise?
+                int[] results = IntStream.range(0, nSamples).map(i -> travelTimeMinutes).toArray();
+                try {
+                    output.writePixel(x, y, results);
+                } catch (IOException e) {
+                    // can't happen as we're not using a file system backed output
+                    throw new RuntimeException(e);
+                }
+            }
+        } else {
+            // we always walk to egress from transit, but we may have a different access mode.
+            // if the access mode is also walk, the pointset's linkage cache will return two references to the same
+            // linkage below
+            // TODO use directMode? Is that a resource limiting issue?
+            // also gridcomputer uses accessMode to avoid running two street searches
+            LinkedPointSet linkedDestinationsAccess = destinations.link(network.streetLayer, accessMode);
+            LinkedPointSet linkedDestinationsEgress = destinations.link(network.streetLayer, StreetMode.WALK);
+
+            if (!request.request.directModes.equals(request.request.accessModes)) {
+                LOG.warn("Disparate direct modes and access modes are not supported in analysis mode.");
+            }
+
+            // Perform street search to find transit stops and non-transit times.
+            StreetRouter sr = new StreetRouter(network.streetLayer);
+            sr.streetMode = accessMode;
+            sr.profileRequest = request.request;
+            sr.distanceLimitMeters = 2000; // TODO hardwired same as gridcomputer
+            sr.setOrigin(request.request.fromLat, request.request.fromLon);
+            sr.dominanceVariable = StreetRouter.State.RoutingVariable.DISTANCE_MILLIMETERS;
+            sr.route();
+
+            // Get the travel times to all stops reached in the initial on-street search. Convert distances to speeds.
+            int offstreetTravelSpeedMillimetersPerSecond = (int) (request.request.getSpeed(accessMode) * 1000);
+
+            TIntIntMap accessTimes = sr.getReachedStops();
```

Comment:
This map does not contain access times. The dominance variable is millimeters, so this contains access distances in millimeters.